deps/tools/common.mk

```diff
@@ -40,16 +40,14 @@ CMAKE_COMMON += -DCMAKE_RC_COMPILER="$$(which $(CROSS_COMPILE)windres)"
 endif
 endif
 
-ifneq (,$(findstring $(OS),Linux FreeBSD))
-INSTALL_RPATH := "\$$ORIGIN"
-# Part of the FreeBSD libgcc_s kludge
+# NOTE: Do not make RPATH changes in CMAKE_COMMON on platforms other than FreeBSD, since
+# it will make its way into the LLVM build flags, and LLVM is picky about RPATH (though
+# apparently not on FreeBSD). Ref PR #22352
 ifeq ($(OS),FreeBSD)
 ifneq ($(GCCPATH),)
-INSTALL_RPATH := "\$$ORIGIN:$(GCCPATH)"
+CMAKE_COMMON += -DCMAKE_INSTALL_RPATH="\$$ORIGIN:$(GCCPATH)"
```

Comment:
This actually ends up duplicating the RPATH specification on FreeBSD, though I nuked everything, rebuilt from scratch, and tested everything, and things are fine.